engine/src/w32spec.cpp

```diff
@@ -455,33 +455,52 @@ char *MCS_get_canonical_path(const char *path)
 {
 	char *t_path = NULL;
 	char *t_curdir = NULL;
+	char *t_livecode_path = NULL;
+	char *t_livecode_path_start;
 
 	if (path == NULL)
 		return NULL;
+	
+	// SN-2015-09-04: [[ Bug 15814 ]] Backslashes are not valid filename chars
+	//  on Windows, so we can accept Windows paths using backslashes (for the 
+	//  people who are still using them in LiveCode scripts).
+	bool t_use_backslashes;
+	t_use_backslashes = false;
+	uint4 i = 0;
+	while (path[i] != NULL && !t_use_backslashes)
+		t_use_backslashes = path[i++] == '\\';
+	
+	t_livecode_path = strclone(path);
+	t_livecode_path_start = t_livecode_path;
 
-	if (path[0] == '/' && path[1] != '/')
+	if (t_use_backslashes)
```

Comment:
Wouldn't it be better to map all / -> \, then check for absoluteness and make absolute if necessary, then map \ to /? (I'm not clear what the use_backslashes is for)